knowledge-base/kb-pdfviewer-viewport-snapshot.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Creating a ViewPort snapshot in RadPdfViewer
+description: Create a snapshot of the visible part of a PDF document in the RadPdfViewer.
+type: how-to
+page_title: How to create a ViewPort snapshot in RadPdfViewer
+slug: kb-pdfviewer-viewport-snapshot
+position: 0
+tags: pdf, pdfviewer, viewport, snapshot
+ticketid: 1512091
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product Version</td>
+	    	<td>2021.1.319</td>
+	    </tr>
+	    <tr>
+            <td>Product</td>
+            <td>RadPDFViewer for WPF</td>
+	    </tr>
+    </tbody>
+</table>
+
+## Description
+
+How to create a snapshot of the visible part (ViewPort) of a **PDF** document loaded in the **RadPdfViewer**.
+
+## Solution
+
+The first step is to create a custom [IUILayer](https://docs.telerik.com/devtools/wpf/api/telerik.windows.documents.fixed.ui.layers.iuilayer), which will help us to obtain the ViewPort and store it in the _VisiblePagesToViewport_ dictionary.
```

Comment:
The first step is to create a custom layer by implementing the [IUILayer](https://docs.telerik.com/devtools/wpf/api/telerik.windows.documents.fixed.ui.layers.iuilayer) interface. The custom implementation will help us to obtain the viewport and store it in the _VisiblePagesToViewport_ dictionary.